internal/manifest/parser/template_to_module.go

```diff
@@ -203,27 +207,44 @@ func getLocalizedImagesFromDescriptor(descriptor *types.Descriptor) []string {
 	return localizedImages
 }
 
-func translateLayersAndMergeIntoManifest(manifest *v1beta2.Manifest, layers img.Layers) error {
+func translateLayersAndMergeIntoManifest(manifest *v1beta2.Manifest, layers img.Layers, repo string) error {
 	for _, layer := range layers {
-		if err := insertLayerIntoManifest(manifest, layer); err != nil {
+		if err := insertLayerIntoManifest(manifest, layer, repo); err != nil {
 			return fmt.Errorf("error in layer %s: %w", layer.LayerName, err)
 		}
 	}
 	return nil
 }
 
-func insertLayerIntoManifest(manifest *v1beta2.Manifest, layer img.Layer) error {
+func insertLayerIntoManifest(manifest *v1beta2.Manifest, layer img.Layer, ociRepoFromConfig string) error {
 	switch layer.LayerName {
 	case v1beta2.DefaultCRLayer:
 		// default CR layer is not relevant for the manifest
 	case v1beta2.ConfigLayer:
-		imageSpec, err := layer.ConvertToImageSpec()
+		imageSpec, err := layer.ConvertToImageSpec(ociRepoFromConfig)
 		if err != nil {
 			return fmt.Errorf("error while parsing config layer: %w", err)
 		}
 		manifest.Spec.Config = imageSpec
 	case v1beta2.RawManifestLayer:
-		installRaw, err := layer.ToInstallRaw()
+		ociImage, ok := layer.LayerRepresentation.(*img.OCI)
+		if !ok {
+			return fmt.Errorf("%w: actual type: %T", ErrConvertingToImgOCI, layer.LayerRepresentation)
+		}
+
+		// For fetching data from the OCI registry use the repo from the global config
+		// instead of the one from the layer (it is the same as in the ComponentDescriptor).
+		// These two values may be different and the explicitly configured one is safer to use,
+		// as it is known to be reachable.
+		// After all, we've been able to read the ComponentDescriptor using it.
+		ociImageCopy := img.OCI{
+			Repo: ociRepoFromConfig,
+			Name: ociImage.Name,
+			Ref:  ociImage.Ref,
+			Type: ociImage.Type,
+		}
```

Comment:
Don't understand the need for it. In `imageSpec, err := layer.ConvertToImageSpec(ociRepoFromConfig)`, we already overwrite the repo part, why creating another copy where we again set the repo explicitly?

Reply:
I indeed got confused here and didn't understand correctly on my side. Thanks for clarifying!